### PR TITLE
Update and Go-ify docs for Settings.

### LINKED ISSUE #42: Update the docs for Settings.GradientAbsTol

### PR DESCRIPTION
**Pull request by vladimir-ch**
Also, fix and reformat comment for resize().

Closes #42

### CODE REVIEW

**Comment by vladimir-ch** [comment]
Why is this coveralls thing constantly failing?

**Comment by jonlawlor** [comment]
The returned message is:
```
Bad response status from coveralls: 503 - <html><body><h1>503 Service Unavailable</h1>
No server is available to handle this request.
</body></html>
```
And if I visit their website at https://coveralls.io/ I get the same 503.  They're having an outage.

I've considered having travis not fail if the coveralls code fails, but it would make it very easy for us to not notice that things weren't working on coveralls for a long period (due to issues such as this one).

This is the first outage that I've seen in ~9 months of watching.  If it is unacceptably long, or if they keep having them, we should probably eliminate it altogether.  @btracey @kortschak any thoughts as to what that point would be?

**Comment by jonlawlor** [comment]
Coveralls.io is back up and I reran the builds. Things look ok.

**Comment by jonlawlor** [comment]
Coveralls.io is back up and I reran the builds. Things look ok.

**Comment by vladimir-ch** [comment]
Great, thanks for looking into it, @jonlawlor . I often (my subjective impression) see a failure with coveralls with the 1.4.1 build, but it seems that this was not related to it.

**Comment by jonlawlor** [comment]
I get notified when there are build errors but I don't usually check them all - I assume that it is a part of normal development.  The next time you see inconsistent outcomes, where some versions of go pass and others fail, I'd appreciate it if you'd ping me with github's @ name thing.  I'll be sure to take a look.

**Comment by jonlawlor** [comment]
Ah, I've just found an example while getting ready to change the 1.4.1 builds to 1.4.2.

File: types.go [hunk 1]
```diff
@@ -121,36 +121,51 @@ type functions struct {
...
+	// FunctionAbsTol is the threshold for acceptably small values of the
+	// objective function. FunctionAbsoluteConvergence status is returned if
+	// the objective function is less than this value.
+	// The default value is -inf.
 	FunctionAbsTol float64
 
-	// Loosely, converge if the 'average' value of the gradient is less than this
-	// value. Specifically, converge if ||grad||_2 / sqrt(len(grad)) is less than
-	// this value.
+	// GradientAbsTol determines the accuracy with which the minimum is to be
```
Comment (btracey) [inline types.go]: to which the minimum is found.

File: types.go [hunk 2]
```diff
@@ -121,36 +121,51 @@ type functions struct {
...
 	// If it equals zero, this setting has no effect.
+	// The default value is 0.
 	MajorIterations int
 
-	// Converge if the duration of the run is longer than this value. Runtime
-	// is only checked at iterations of the optimizer. If it equals zero,
-	// this setting has no effect.
+	// Runtime is the maximum runtime allowed. RuntimeLimit status is returned
+	// if the duration of the run is longer than this value. Runtime is only
+	// checked at iterations of the optimizer.
```
Comment (btracey) [inline types.go]: this should be "iterations of the Method", right?

**Comment by btracey** [comment]
LGTM after the minor fixes.

**Comment by vladimir-ch** [comment]
Fixed both comments.

**Comment by kortschak** [comment]
@jonlawlor can you make it retry on failure (once only) for coveralls?

**Pull request merged**